src/libPMacc/include/cuSTL/algorithm/kernel/run-time/Foreach.hpp

```diff
@@ -68,6 +92,8 @@ namespace RT
         /* C0 c0_shifted = c0(p_zone.offset); ...; CN cN_shifted = cN(p_zone.offset); */                              \
         BOOST_PP_REPEAT(N, SHIFT_CURSOR_ZONE, _)                                                                    \
                                                                                                                     \
+        if(this->_blockDim == math::Size_t<DIM3>::create(0))                                                           \
```

Comment:
Is the `getBestCudaBlockDim` method only called if the user defines the block size as `(0,0,0)`? 
If so please add somewhere a documentation that the user knows about this feature.